src/server/dfly_main.cc

```diff
@@ -420,16 +427,15 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
     const char* interface_addr = admin_bind.empty() ? nullptr : admin_bind.c_str();
     const std::string printable_addr =
         absl::StrCat("admin socket ", interface_addr ? interface_addr : "any", ":", admin_port);
-    Listener* admin_listener = new Listener{Protocol::REDIS, &service};
+    auto admin_listener = std::make_unique<Listener>(Protocol::REDIS, &service);
     admin_listener->SetAdminInterface();
-    error_code ec = acceptor->AddListener(interface_addr, admin_port, admin_listener);
+    error_code ec = acceptor->AddListener(interface_addr, admin_port, admin_listener.get());
```

Comment:
Not the best thing in the world that two `unique_ptr` point to the same object but also not the end of the world because at least now we don't have raw deletes and the semantics here do not lead to UB under any circumstance